# Imports

In [1]:
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
from ipynb.fs.defs.problem1 import estimate_capm


## Load in Fama French Portfolio Data

In [3]:
# Read the file, skipping header text rows and limiting to monthly data
ff = pd.read_csv("F-F_Research_Data_Factors 3.CSV", skiprows=3, nrows=1182)

# Rename the date column for clarity
ff.rename(columns={"Unnamed: 0": "Date"}, inplace=True)

# Convert Date to string to extract Year and Month
ff['Date'] = ff['Date'].astype(str)
ff['Year'] = ff['Date'].str[:4].astype(int)
ff['Month'] = ff['Date'].str[4:].astype(int)

# Convert other values to floats
ff['Mkt-RF'] = ff['Mkt-RF'].astype(float)
ff['RF'] = ff['RF'].astype(float)

# Pull the columns we want
ff = ff[['Year', 'Month', 'Mkt-RF', 'RF']]

## Load in Strategies Dataset

In [4]:
# Read the file
ss = pd.read_csv("ps1_strategies.csv")

# Convert Date to string to extract Year and Month
ss['date'] = ss['date'].astype(str)
ss['Year'] = ss['date'].str[:4].astype(int)
ss['Month'] = ss['date'].str[4:].astype(int)

# Make other values floats
ss['CA'] = ss['CA'].astype(float)

# Pull the columns we want
ss = ss[['Year', 'Month', 'CA']]

# Part (a)

In [7]:
results_lbha = estimate_capm(ss, ff, 'LBHA')

print("CAPM Results for LBHA Strategy")
print(f"Alpha: {results_lbha['alpha']:.4f}")
print(f"Beta: {results_lbha['beta']:.4f}")
print(results_lbha['summary'])

CAPM Results for LBHA Strategy
Alpha: 0.2729
Beta: 0.0067
                            OLS Regression Results                            
Dep. Variable:                  RP-RF   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.002
Method:                 Least Squares   F-statistic:                   0.08125
Date:                Thu, 10 Apr 2025   Prob (F-statistic):              0.776
Time:                        00:42:27   Log-Likelihood:                -853.10
No. Observations:                 397   AIC:                             1710.
Df Residuals:                     395   BIC:                             1718.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------

Based on these regression results, LBHA does indeed show positive and statistically significant alpha (0.2729, p=0.01) and a very low market beta (0.0067). Because alpha is both positive and significant, the strategy is generating returns above and beyond what would be explained by its market exposure. In other words, LBHA exhibits risk-adjusted outperformance relative to the CAPM market benchmark, which implies that it **beats the market** under this model.

# Part (b)

# Part (c)